OS_CYGWIN is used for cygpath and other Win32 assumptions that we don't want.

--- origsrc/boost_1_37_0/tools/jam/src/jam.h	2007-11-19 13:25:21.000000000 -0600
+++ src/boost_1_37_0/tools/jam/src/jam.h	2009-06-11 17:35:26.541005500 -0500
@@ -244,7 +244,7 @@
 # endif
 # if defined(__cygwin__) || defined(__CYGWIN__)
 # define OSMINOR "OS=CYGWIN"
-# define OS_CYGWIN
+# define OS_CYGWIN17
 # endif
 # if defined(__FreeBSD__) && !defined(__DragonFly__)
 # define OSMINOR "OS=FREEBSD"
